snappy/snapp.go

```diff
@@ -1943,11 +1960,13 @@ func (s *SnapUbuntuStoreRepository) Updates() (parts []Part, err error) {
 		return nil, err
 	}
 
+	parts := make([]Part, 0, len(updateData))
 	for _, pkg := range updateData {
 		current := ActiveSnapByName(pkg.Name)
 		if current == nil || current.Version() != pkg.Version {
-			snap := NewRemoteSnapPart(pkg)
-			parts = append(parts, snap)
+			if part, err := s.detail(pkg.Name, pkg.Origin, pkg.Channel); err == nil {
```

Comment:
Maybe a small comment here why we call "detail()" again instead of just using the "pkg" we got from the store to avoid my future self being confused when I forgot about this branch?